app/services/workflow_update_service.rb

```diff
@@ -6,27 +6,10 @@ def initialize(workflow_id)
   end
 
   def update(options)
-    original_group_refs = workflow.group_refs
-    current_group_refs = options[:group_refs]
-
-    if current_group_refs
-      removed_group_refs = original_group_refs - current_group_refs
-      added_group_refs = current_group_refs - original_group_refs
-
-      aps = AccessProcessService.new
-
-      if added_group_refs.any?
-        ContextService.new(Insights::API::Common::Request.current.to_h.transform_keys(&:to_s)).as_org_admin do
-          aps.add_resource_to_groups(workflow.id, added_group_refs)
-        end
-      end
-
-      if removed_group_refs.any?
-        ContextService.new(Insights::API::Common::Request.current.to_h.transform_keys(&:to_s)).as_org_admin do
-          aps.remove_resource_from_groups(workflow.id, removed_group_refs)
-        end
+    options[:access_control_entries] =
+      (options[:group_refs] || []).collect do |uuid|
+        AccessControlEntry.new(:group_uuid => uuid, :permission => 'approve')
```

Comment:
We should remove the old `AccessControlEntry` here, right? Otherwise the following lookup based on local acls will return out of sync records.

Reply:
The `update` method automatically removes outdated associations.

Reply:
👍